circuits/integration/mpc_gadgets/bits.rs

```diff
@@ -110,80 +99,51 @@ fn test_bit_add(test_args: &IntegrationTestArgs) -> Result<(), String> {
-        .open_and_authenticate()
-        .map_err(|err| format!("Error opening party0's random number: {:?}", err))?;
-
-    let random_number2 = test_args
-        .borrow_fabric()
-        .allocate_private_u64(1 /* owning_party */, my_random_number)
-        .map_err(|err| format!("Error sharing random number: {:?}", err))?
-        .open_and_authenticate()
-        .map_err(|err| format!("Error opening party1's random number: {:?}", err))?;
-
-    let expected_result = scalar_to_u64(&(random_number1.to_scalar() + random_number2.to_scalar()));
-    let expected_bits = u64_to_bits_le(expected_result);
-
-    let all_equal = result
-        .iter()
-        .zip(expected_bits.iter().cloned())
-        .all(|(res, expected)| res.to_scalar().eq(&Scalar::from(expected)));
-
-    if !all_equal {
-        return Err(format!(
-            "Expected: {:?}, got {:?}",
-            expected_bits,
-            result
-                .iter()
-                .map(|bit| scalar_to_u64(&bit.to_scalar()))
-                .collect::<Vec<_>>(),
-        ));
-    }
-
-    Ok(())
+    let random_number1 =
+        await_result(fabric.share_plaintext(Scalar::from(my_random_number), PARTY0));
+    let random_number2 =
+        await_result(fabric.share_plaintext(Scalar::from(my_random_number), PARTY1));
+
+    let expected_result = scalar_to_u64(&(random_number1 + random_number2));
+    let expected_bits = u64_to_bits_le(expected_result)
+        .into_iter()
+        .map(Scalar::from)
+        .collect_vec();
+
+    assert_scalar_batch_eq(&expected_bits, &result)
 }
 
 /// Tests that getting the bits of 0 returns all zeros
 fn test_bits_le_zero(test_args: &IntegrationTestArgs) -> Result<(), String> {
-    let shared_value = test_args
-        .borrow_fabric()
-        .allocate_private_u64(0 /* owning_party */, 0 /* value */)
-        .map_err(|err| format!("Error sharing value: {:?}", err))?;
-
-    let shared_bits = to_bits_le::<250, _, _>(&shared_value, test_args.mpc_fabric.clone())
-        .map_err(|err| format!("Error computing bits(0): {:?}", err))?
-        .iter()
-        .map(|bit| bit.open_and_authenticate().unwrap())
-        .collect::<Vec<_>>();
-
-    check_equal_vec(&shared_bits, &vec![0u64; shared_bits.len()])
+    let fabric = &test_args.mpc_fabric;
+    let shared_zero = fabric.zero_authenticated();
+
+    let shared_bits = to_bits_le::<250>(&shared_zero, test_args.mpc_fabric.clone());
+    let shared_bits_open = await_result_batch_with_error(
+        &AuthenticatedScalarResult::open_authenticated_batch(&shared_bits),
+    )?;
+
+    assert_scalar_batch_eq(&shared_bits_open, &vec![Scalar::zero(); shared_bits.len()])
 }
 
 /// Tests the to_bits_le gadget
 fn test_to_bits_le(test_args: &IntegrationTestArgs) -> Result<(), String> {
+    let fabric = &test_args.mpc_fabric;
     let value = 119;
 
     // The parties share the value 10 with little endian byte representation 0b0101
```

Comment:
nit: either remove this comment or update it:

```suggestion
    // The parties share the value 119 with little endian byte representation 0b11101110
```